### PR TITLE
feat(modals): #MED-310 add modals to create, edit and delete pin resource

### PR DESCRIPTION
**Pull request by Mattox0**
## Describe your changes

Add 3 modals for create, delete or update a pin resource.
image

## Checklist tests

- Check if we can use 3 modals

## Issue ticket number and link

[MED-310](https://jira.support-ent.fr/browse/MED-310)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/routes/app/index.tsx [hunk 1]
```diff
@@ -256,43 +271,47 @@ export const App = () => {
...
+              <div className="empty-state">
+                <img
+                  src="/mediacentre/public/img/empty-state.png"
+                  alt="empty-state"
+                  className="empty-state-img"
+                />
+                <span className="empty-state-text">
+                  {t("mediacentre.ressources.empty")}
+                </span>
+              </div>
```
Comment (JinDouang) [inline frontend/src/routes/app/index.tsx:301]: tu peux pas faire un composant pour l'empty state ?

File: frontend/src/components/pins-carousel/PinsCarousel.scss [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  .splide__arrow--next {
+    right: 0px;
+  }
+
+  .med-pagination {
+    margin: 2rem 0 0 0;
+  }
+
+  .splide__pagination__page {
+    background: #d9d9d9;
```
Comment (JinDouang) [inline frontend/src/components/pins-carousel/PinsCarousel.scss:51]: tu peux pas variabiliser cette valeur ?

**Review by JinDouang — CHANGES_REQUESTED** [review]

**Review by JinDouang — APPROVED** [review]

File: frontend/src/components/infinite-scroll-list/InfiniteScrollList.tsx [hunk 3]
```diff
@@ -12,15 +12,13 @@ import { Resource } from "~/model/Resource.model";
 interface InfiniteScrollListProps {
   redirectLink: string;
   allResourcesDisplayed: Resource[] | null;
-  setAlertText: (alertText: string) => void;
-  refetchData: () => void;
+  refetchSearch: () => void;
```
Comment (niko-amosse) [inline frontend/src/components/infinite-scroll-list/InfiniteScrollList.tsx:15]: remettre refetchData

File: frontend/src/components/modals/create-pins/CreatePins.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      <Modal.Body>
+        <div className="med-modal-container">
+          <div className="med-modal-image">
+            <img
+              src={modalResource?.image}
+              alt="Resource"
+              className="med-image"
+              onError={({ currentTarget }) => {
+                currentTarget.onerror = null;
+                currentTarget.src = "/mediacentre/public/img/no-avatar.svg";
```
Comment (niko-amosse) [inline frontend/src/components/modals/create-pins/CreatePins.tsx:104]: mettre la nouvelle image de james

File: frontend/src/components/modals/edit-pins/EditPins.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      <Modal.Body>
+        <div className="med-modal-container">
+          <div className="med-modal-image">
+            <img
+              src={modalResource?.image}
+              alt="Resource"
+              className="med-image"
+              onError={({ currentTarget }) => {
+                currentTarget.onerror = null;
+                currentTarget.src = "/mediacentre/public/img/no-avatar.svg";
```
Comment (niko-amosse) [inline frontend/src/components/modals/edit-pins/EditPins.tsx:108]: idem nouvelle image de james

File: frontend/src/components/infinite-scroll-list/InfiniteScrollList.tsx [hunk 6]
```diff
@@ -12,15 +12,13 @@ import { Resource } from "~/model/Resource.model";
...
 }
 
 export const InfiniteScrollList: React.FC<InfiniteScrollListProps> = ({
   redirectLink,
   allResourcesDisplayed,
-  setAlertText,
-  refetchData,
+  refetchSearch,
```
Comment (niko-amosse) [inline frontend/src/components/infinite-scroll-list/InfiniteScrollList.tsx:21]: idem

File: frontend/src/components/infinite-scroll-list/InfiniteScrollList.tsx [hunk 7]
```diff
@@ -102,8 +100,7 @@ export const InfiniteScrollList: React.FC<InfiniteScrollListProps> = ({
                   <SearchCard
                     searchResource={searchResource}
                     link={searchResource.link ?? searchResource.url ?? "/"}
-                    setAlertText={setAlertText}
-                    refetchData={refetchData}
+                    refetchSearch={refetchSearch}
```
Comment (niko-amosse) [inline frontend/src/components/infinite-scroll-list/InfiniteScrollList.tsx:103]: idem

File: frontend/src/components/pins-carousel/PinsCarousel.tsx [hunk 8]
```diff
@@ -50,7 +50,7 @@ export const PinsCarousel: React.FC<PinsCarouselProps> = ({ pins }) => {
         <SplideTrack className="med-splide-track">
           {pins.map((pin, index) => (
             <SplideSlide key={index} className="">
-              <PinsCarouselCard pin={pin} />
+              <PinsCarouselCard pin={pin} link={pin?.link ?? pin?.url ?? ""}/>
```
Comment (niko-amosse) [inline frontend/src/components/pins-carousel/PinsCarousel.tsx:53]: on peut faire ça direct dans le composant non ? (pour le link)

File: frontend/src/components/search-card/SearchCard.tsx [hunk 9]
```diff
@@ -23,21 +26,21 @@ import {
 interface SearchResourceProps {
   searchResource: SearchResource;
   link: string;
-  setAlertText: (arg: string, type: AlertTypes) => void;
-  refetchData: () => void;
+  refetchSearch: () => void;
```
Comment (niko-amosse) [inline frontend/src/components/search-card/SearchCard.tsx:29]: idem

File: frontend/src/components/search-card/SearchCard.tsx [hunk 10]
```diff
@@ -23,21 +26,21 @@ import {
...
 }
 
 export const SearchCard: React.FC<SearchResourceProps> = ({
   searchResource,
   link,
-  setAlertText,
-  refetchData,
+  refetchSearch,
```
Comment (niko-amosse) [inline frontend/src/components/search-card/SearchCard.tsx:35]: idem

File: frontend/src/routes/resources/index.tsx [hunk 11]
```diff
@@ -87,8 +87,7 @@ export const ResourcePage: React.FC = () => {
             <InfiniteScrollList
               redirectLink="/resources"
               allResourcesDisplayed={allResourcesDisplayed}
-              setAlertText={setAlertText}
-              refetchData={refetchSearch}
+              refetchSearch={refetchSearch}
```
Comment (niko-amosse) [inline frontend/src/routes/resources/index.tsx:90]: idem

File: frontend/src/routes/search/index.tsx [hunk 12]
```diff
@@ -102,8 +101,7 @@ export const Search: React.FC = () => {
             <InfiniteScrollList
               redirectLink="/search"
               allResourcesDisplayed={allResourcesDisplayed}
-              setAlertText={setAlertText}
-              refetchData={refetchSearch}
+              refetchSearch={refetchSearch}
```
Comment (niko-amosse) [inline frontend/src/routes/search/index.tsx:104]: idem

File: frontend/src/routes/textbook/index.tsx [hunk 13]
```diff
@@ -102,8 +101,7 @@ export const TextbookPage: React.FC = () => {
             <InfiniteScrollList
               redirectLink="/textbook"
               allResourcesDisplayed={allResourcesDisplayed}
-              setAlertText={setAlertText}
-              refetchData={() => {
+              refetchSearch={() => {
```
Comment (niko-amosse) [inline frontend/src/routes/textbook/index.tsx:104]: idem

**Review by niko-amosse — APPROVED** [review]

File: frontend/src/components/pins-carousel/PinsCarousel.tsx (hunk 8 above)
Comment (Mattox0) [reply frontend/src/components/pins-carousel/PinsCarousel.tsx]: en soit oui, mais c'est plus pratique de le faire directement dans les props, sinon tu devrais faire le if partout dans la card

**Pull request merged**